### ISSUE TITLE
Documentation needed for how to do code verification and code quality 

### ISSUE THREAD

**Issue by Griffintaur** [PlasmaPy/PlasmaPy#386]
I couldn't find below-mentioned points either on Github or website.
Documentation missing points :
- How to do verify code changes or play with the code [for developers/contributors]
- How to do verify code quality and follow style guidelines before contributing
- Things to do before you make a pull request
- How/ Where to write the unit test to ensure code coverage.

**Comment by StanczakDominik** [comment]
@Griffintaur Thanks for pointing this out! The only way our docs are going to get good is by people actually using them, so I'm really grateful for trying. I'm gonna discuss your points one by one:

> How to do verify code changes or play for developers / contributors

If you mean local testing here, that's included: http://docs.plasmapy.org/en/master/development/testing_guide.html . If you mean something else, please elaborate :)

> How to do verify code quality and follow style guidelines before contributing

Yeah, that's something we have to improve [here](http://docs.plasmapy.org/en/master/development/code_guide.html#coding-style). That can probably be worked in along #357 . The docs on this are pretty incomplete, that's totally on us.

> Things to do before you make a pull request

Well... the logic we've been favouring is "make a branch at your fork and go, so that you can use the tools provided by us and GitHub, such as automated tests, @pep8speaks, code review, and so on.". Perhaps that should be explicitly mentioned?

> How/ Where to write unit test to ensure code coverage.

Well, this would be http://docs.plasmapy.org/en/master/development/testing_guide.html but I think we could add something like:
* write docstring examples to show how your function works for common use cases AND to make a persistent record of the function's behavior between commits (if the behavior changes, you'll know by the doctest outputs changing)
* write full scale tests in a `tests` subdirectory at the same location as your code. Compare numerical results to known values, provide sources for those.

**Comment by lemmatum** [comment]
@StanczakDominik :laughing: 
I like your list and especially that you provided links to places in the docs that already answer some of these questions. It'll make linking and organizing things easier. I don't see any conflicts between our lists, so it should be easy to just combine all the ideas for great good!

I think I'd also add:
* mention when you are working on an issue by commenting within that issue
* open PRs early so everyone see what you are working on and can provide feedback
* commit often and use descriptive commit messages
* Add a comment/label to your PR when it is "Ready for Review"

**Comment by StanczakDominik** [comment]
@lemmatum  to your last point:
* Put `[WIP]` (standing for `Work in Progress`) in your PR's title while you don't want it merged just yet and you're just looking for feedback.
